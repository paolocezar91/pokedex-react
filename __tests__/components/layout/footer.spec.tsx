import Footer from '@/components/layout/footer';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';

jest.mock('next/router', () => jest.requireActual('next-router-mock'));
jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key,
    i18n: {
      language: 'en-US',
      changeLanguage: jest.fn()
    }
  }),
}));


describe('Footer', () => {
  it('should update select value when language changes', () => {
    const mockChangeLanguage = jest.fn();

    // eslint-disable-next-line @typescript-eslint/no-require-imports
    jest.spyOn(require('react-i18next'), 'useTranslation').mockReturnValue({
      i18n: {
        language: 'en-US',
        changeLanguage: mockChangeLanguage,
      },
      t: (key: string) => key,
    });

    render(<Footer />);

    const langSelect = screen.getByTestId('lang') as HTMLSelectElement;

    // Verify initial value
    expect(langSelect.value).toBe('en-US');

    // Simulate language change
    fireEvent.change(langSelect, { target: { value: 'pt-BR' } });

    // Verify the changeLanguage function was called
    expect(mockChangeLanguage).toHaveBeenCalledWith('pt-BR');
  });

  it('should match snapshot with no valid description', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});