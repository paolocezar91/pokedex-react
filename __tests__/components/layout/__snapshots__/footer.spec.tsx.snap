// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot with no valid description 1`] = `
<DocumentFragment>
  <div
    class="footer container w-full fixed bottom-0 flex justify-between border-solid border-t-2 border-(--pokedex-red-light) bg-(--pokedex-red) py-2 px-4"
  >
    <div
      class="flex items-center text-xs text-white"
    >
      <p>
        By 
        <a
          class="underline"
          href="https://github.com/paolocezar91/"
          target="_blank"
        >
          Paolo Pestalozzi
        </a>
         with 
        <a
          class="underline"
          href="https://pokeapi.co/"
          target="_blank"
        >
          PokeAPI
        </a>
         and 
        <a
          class="underline"
          href="http://nextjs.org/"
          target="_blank"
        >
          Next.js
        </a>
        .
      </p>
    </div>
    <div
      class="next flex-1 text-right"
    >
      <div
        class="tooltip-container"
      >
        <div
          class="tooltip-trigger"
        >
          <select
            class="border-solid border-2 border-black text-black bg-white py-2 rounded text-xs"
            data-testid="lang"
            id="lang"
          >
            <option
              value="en-US"
            >
              en-US
            </option>
            <option
              value="pt-BR"
            >
              pt-BR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
